File: despatch.Domain/Models/Default/Request/RoomRequest.cs
using despatch.Domain.Models.Enums;
using System.Collections.Generic;

namespace despatch.Domain.Models.Default.Request
{
    public class RoomRequest : BaseEntityRequest
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public RoomType Type { get; set; }
        public RoomPrivacyType PrivacyType { get; set; }

        public string CreatorId { get; set; }
        public string LastMessageId { get; set; }

        public UserRequest Creator { get; set; }
        public MessageRequest LastMessage { get; set; }

        public virtual ICollection<UserRoomsRequest> UserRooms { get; set; }
        public virtual ICollection<ImageRequest> Images { get; set; }
    }
}

File: despatch.WEB/app/src/swagger/models/room-request.ts
/* tslint:disable */
import { BaseEntityRequest } from './base-entity-request';
import { UserRequest } from './user-request';
import { ImageRequest } from './image-request';
import { MessageRequest } from './message-request';
import { RoomPrivacyType } from './room-privacy-type';
import { RoomType } from './room-type';
import { UserRoomsRequest } from './user-rooms-request';
export interface RoomRequest extends BaseEntityRequest{
  creator?: UserRequest;
  creatorId?: string;
  description?: string;
  images?: Array<ImageRequest>;
  lastMessage?: MessageRequest;
  lastMessageId?: string;
  name?: string;
  privacyType: RoomPrivacyType;
  type: RoomType;
  userRooms?: Array<UserRoomsRequest>;
}
